### DISCUSSION TITLE
Add option to include stdout, stderr and exitcode to steps output

### DISCUSSION THREAD

**Discussion by AlissonRS** [community/community#83102] [Actions]
### Select Topic Area

Product Feedback

### Body

Getting the output of a certain step is a common task in GitHub Actions, yet very tedious.

Please refer to: https://stackoverflow.com/questions/59191913/how-do-i-get-the-output-of-a-specific-step-in-github-actions

One idea is "echoing" the commdn to GITHUB_OUTPUT, which makes the step to succeed even if the actual command failed as it's masked by the echo command.

There are workarounds for this, but they make the step way more complex than it should be.

Someone even went through the trouble of creating a custom action as a replacement to the run command: https://github.com/mathiasvr/command-output

People have asked for this feature before (https://github.com/actions/runner/issues/919) but they were shutdown without even a proper discussion.

One guy mentioned about the runner's memory, but that's a non-issue, we can simply make this feature optional, it could look like this:
    
    steps:
      - name: Build
        id: build
        stdout: true
        stderr: true
        exitcode: true
        run: yarn build
      - name: output build result
        run: echo ${{ steps.build.outputs.stdout }}

...and make these things optional and false by default. This example looks silly, but we could use the stdout/stderr/exitcode to add a comment to a pull request, or send a notification on Slack/Discord.
        
This feature would not necessarily create a memory problem as people are already doing this by appending stuff to GITHUB_OUTPUT, it's just inconvenient because the `echo` command hides the command exit code and make the step succeed.

**Comment by ma7dev** [comment]
+1 for this feature!

GitHub Actions can be used for a lot of things, from ensuring that codebase standards are applied to doing a chain of automated work. One usage of GitHub Actions is to automate experimentations using GitHub Actions. I have set up a full example of how to do [Continuous Experimentation using GitHub Actions](https://github.com/ma7dev/simple-ce-example). It would be useful to have `stdout`, `stderr` and `exitcode` to debug issues with jobs when they occur.

One more feature request: PLEASE HAVE THE ABILITY TO RUN JOBS LOCALLY FOR TESTING!!!!!! :(

**Comment by mdeweerd** [comment]
I just discovered that capturing output cross-platform (matrix) tests can be a pain as "tee" exists on Linux but not on Windows for instance.

There are two options IMHO:
- make the output(s) available in a variable;
- make the output(s) available as the path to the filename where it is stored (and accessible from the next steps).

Note: output(s) because of stdout and stderr.

**Comment by awittzb** [comment]
While workarounds exist for the `run` step, a step that `uses:` another GitHub Action has no way to capture the stdout/stderr output unless the individual Action being used instruments itself to expose these. Nearly all actions do *not* do that.

Built-in support for capturing every step's output would solve this.

# Has Workaround

`run` usage
```yaml
...
steps:
  - run: conda install ...
```

`run` usage w/ stdout capture
```yaml
steps:
  - run: conda install ... | tee conda.txt
  - run: cat conda.txt
```


# No Workaround

`uses` usage
```yaml
...
steps:
  - uses: cycjimmy/semantic-release-action@v4
    id: release
  - name: "Process release stdout"
    ...?
```

⚠️ There is no workaround to capture the stdout from a `uses:` step!

**Reply by david-v-pierce-usps** [reply]
The thing I don't like about "tee" is that it also suppressed the error code if the "conda install" fails.

It can be worked around (just need an intermediate step that does a little output analysis) but I think it's a little annoying, personally.
